apps/web/src/components/NotificationList/mutations.ts

```diff
@@ -47,12 +47,12 @@ export function useMarkAsRead(
     return executeMutation({ id })
       .then((res) => {
         if (!res.data?.markNotificationAsRead) {
-          return Promise.reject();
+          return Promise.reject(new Error(res.error?.toString()));
         }
 
         return Promise.resolve();
       })
-      .catch(() => Promise.reject());
+      .catch(() => Promise.reject(new Error()));
```

Comment:
could we catch the err object here and read the message property from that if available?

Reply:
Yeah, that makes sense :+1: Added in [better error handling for notifications list](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11676/commits/f3db2605e544892fec2a88562a2b5d4ddef441b5)